src/System.Net.Sockets/tests/FunctionalTests/SocketOptionNameTest.cs

```diff
@@ -293,5 +293,64 @@ public void ReuseAddress_Windows(bool? exclusiveAddressUse, bool? firstSocketReu
         {
             ReuseAddress(exclusiveAddressUse, firstSocketReuseAddress, secondSocketReuseAddress, expectFailure);
         }
+
+        [OuterLoop] // TODO: Issue #11345
+        [Theory]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        [InlineData(IPProtectionLevel.EdgeRestricted)]
+        [InlineData(IPProtectionLevel.Restricted)]
+        [InlineData(IPProtectionLevel.Unrestricted)]
+        public void SetIPProtectionLevel_Windows(IPProtectionLevel level)
+        {
+            using (var socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp))
+            {
+                socket.SetIPProtectionLevel(level);
+
+                int result = (int)socket.GetSocketOption(SocketOptionLevel.IP, SocketOptionName.IPProtectionLevel);
+                Assert.Equal(result, (int)level);
+            }
+
+            using (var socket = new Socket(AddressFamily.InterNetworkV6, SocketType.Stream, ProtocolType.Tcp))
+            {
+                socket.SetIPProtectionLevel(level);
+
+                int result = (int)socket.GetSocketOption(SocketOptionLevel.IPv6, SocketOptionName.IPProtectionLevel);
+                Assert.Equal(result, (int)level);
+            }
+        }
```

Comment:
Is there anything more we can/should test about the impact this option has on behavior?  If not, ok.